rclcpp/test/rclcpp/CMakeLists.txt

```diff
@@ -726,6 +726,23 @@ if(TARGET test_rosout_qos)
   target_link_libraries(test_rosout_qos ${PROJECT_NAME})
 endif()
 
+function(test_rosout_subscription_for_rmw_implementation)
+  set(rmw_implementation_env_var RMW_IMPLEMENTATION=${rmw_implementation})
+  ament_add_gmock(test_rosout_subscription${target_suffix} test_rosout_subscription.cpp
+    ENV ${rmw_implementation_env_var}
+  )
+  if(TARGET test_rosout_subscription${target_suffix})
+    target_link_libraries(test_rosout_subscription${target_suffix} ${PROJECT_NAME})
+    ament_target_dependencies(test_rosout_subscription${target_suffix}
+      "rcl_interfaces"
+      "rmw"
+      "rosidl_runtime_cpp"
+      "rosidl_typesupport_cpp"
+    )
+  endif()
+endfunction()
+call_for_each_rmw_implementation(test_rosout_subscription_for_rmw_implementation)
```

Comment:
Honestly, I don't think we need to test this for each RMW implementation.  That is, the point of this test is to test the rosout subscription, not to test out how well the RMWs work.  I would just let this one run with whatever is currently configured to be the default RMW.

Reply:
Updated in https://github.com/ros2/rclcpp/pull/1717/commits/48e13f1b687604a4ecd41ebf3d6c06207ba21caf